Weather App Review

didymus707 opened the pull request:
In this request, I made an api call to openweather API to get weather conditions of a location.
- Errors are being handled
- User can search for weather condition by city name
- Temp can be seen on both celsius and farenheit on click

didymus707 committed:
Added .github/workflow/linters.yml

didymus707 committed:
Installed eslint, stylelint, added stylelintrc and eslintrc.json

didymus707 committed:
Added src/index.js file Added webpack config file

didymus707 committed:
Added output key and value in module.exports in webpack.config

didymus707 committed:
Added css module loader in webpack.config

didymus707 committed:
Added css file Imported css file

didymus707 committed:
Created getData to fetch data from api Installed htmlwebpackplugin and configured it in webpack.config

didymus707 committed:
Added devtool in webpack.config for development

didymus707 committed:
Installed babel for transpiling and added the rule sets

didymus707 committed:
Adjusted babel ruleset Corrected the api address

didymus707 committed:
Installed regenerator-runtime Created processData to process result from api call

didymus707 committed:
Added helper function to set Attributes

didymus707 committed:
Added and exported page.js file Created page elements Imported it in entry point

didymus707 committed:
User can switch between farnheit and celcius

didymus707 committed:
Installed country code lookupAdded p tag to pages

didymus707 committed:
Removed a li tag Looked for country by its Iso Edited conversion function Added a lister to header to switch between units

didymus707 committed:
Added styles

didymus707 committed:
Added a top header

didymus707 committed:
Updated readme

didymus707 committed:
Linter errors fixed

trekab commented on package.json:
```diff
@@ -0,0 +1,42 @@
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "watch": "webpack --watch"
+  },
```
Add the necessary run common commands like `build`, `development` to your `package.json` inside the `scripts` section. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

trekab reviewed (changes requested):
### Changes Required! :recycle: 
Hi @didymus707! Well done working with `async/await` as required on this project. There are a few issues I'd like you to consider though:
- [ ] Please add a toggle control to determine if the result will be shown in Celsius or Fahrenheit. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)) Your current implementation of this feature is easily missed by any user. It took me a while to figure out your toggle functionality. Please explicitly indicate this toggle button. You could alternatively add more descriptive instructions about running your app to the README file indicating how to toggle the temperature.
- [ ] __[Optional]__ Consider deploying your app and adding the live link to your repository. Also, make your set up instructions more explicit in the README.md file. e.g. what does the user do next after installing all the dependencies with the `npm i` command?

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

didymus707 committed:
Removed commented console.log line

didymus707 committed:
Removed label element from page

didymus707 committed:
Added click listener to button

didymus707 committed:
1. Installed moment 2. Added its ruleset in webpack.config 3. Added styles to form element 4. Created mins function to parse time of location

didymus707 committed:
Installed moment-locales-webpack-plugin

didymus707 committed:
1. Added a display property to none class
2. Changed my page structure
3. Added 3 methods(fetcherror, removeNone, checkresult), Added date, Selected 3 elements from page
4. Removed eslint disable line from webpack.config
5. Updated readme

didymus707 committed:
Corrected linting error

MuriloRoque reviewed (approved):
## Status: Approved :green_circle: 

### Hello @didymus707 :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[![Twitter Follow](https://img.shields.io/twitter/follow/MuriloRoquePai1?label=Follow%20Murilo%20on%20Twitter&style=social)](https://twitter.com/MuriloRoquePai1)

didymus707 committed:
Added Styles
